src/Bicep.Core/Emit/EmitConstants.cs

```diff
@@ -9,4 +11,11 @@ public static class EmitConstants
 
     // IMPORTANT: Do not update this API version until the new one is confirmed to be deployed and available in ALL the clouds.
     public const string NestedDeploymentResourceApiVersion = "2022-09-01";
+
+    public const string NestedDeploymentResourceApiVersionWithModuleConfigsEnabled = "2025-03-01";
+
+    public static string GetNestedDeploymentResourceApiVersion(IFeatureProvider features) =>
+        features is { ExtensibilityEnabled: true, ModuleExtensionConfigsEnabled: true }
+            ? NestedDeploymentResourceApiVersionWithModuleConfigsEnabled
+            : NestedDeploymentResourceApiVersion;
```

Comment:
Could you add a TODO comment to note that we should remove this and upgrade `NestedDeploymentResourceApiVersion` to 2025-03-01 once it's been rolled out to all clouds?

Reply:
Done.